### PR TITLE
octopus: Add sparskit variant

### PR DESCRIPTION
**Pull request by iamashwin99**
sparskit is an optional dependency of octopus.
This MR adds sparskit as a variant to the octopus spack package.
A variant of this MR has been tested at https://github.com/fangohr/octopus-in-spack/pull/80 to check that octopus does pick up the sparskit variant.

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
LGTM.  Deferring merge decision to @RemiLacroix-IDRIS and or @fangohr.

File: var/spack/repos/builtin/packages/octopus/package.py [hunk 1]
```diff
@@ -205,6 +211,8 @@ def configure_args(self):
 
         # --with-etsf-io-prefix=
         # --with-sparskit=${prefix}/lib/libskit.a
```
Comment (fangohr) [inline var/spack/repos/builtin/packages/octopus/package.py]:
```suggestion
```
Suggest to delete the comment which presumably was meant to remind us to add the --sparskit variant.

**Review by fangohr — COMMENTED** [review]
Apart from suggestion to remove comment, this looks good to me.

File: var/spack/repos/builtin/packages/octopus/package.py (hunk 1 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/octopus/package.py]: It's obviously correct to remove the now obsolete comment. Committing the suggestion of fangohr.

**Review by bernhardkaindl — DISMISSED** [review]
Approved by @fangohr, obsolete comment is removed and builds fine on Ubuntu 20.04 with it's gcc-10.3.0

**Review by bernhardkaindl — DISMISSED** [review]
Update:
- Initially, adding the "--with-sparskit=%s" % os.path.join(self.spec["sparskit"].prefix.lib, "libskit.a") to args was not done. Added args.append() around it and built it on Ubuntu 20.04 again.
- It had to be rebased on top of the now merged https://github.com/spack/spack/pull/36490

**Comment by bernhardkaindl** [comment]
@spackbot help
@spackbot fix-style

**Comment by bernhardkaindl** [comment]
@spackbot fix style

**Review by bernhardkaindl — APPROVED** [review]
Approving spackbot's style fix ;-)

**Pull request merged**